src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -171,9 +172,24 @@ public static Double parseAmount(String amountStr) throws ParseException {
             logger.fine("ParseException caused by invalid amount or incorrect amount format.");
             throw new ParseException(Messages.MESSAGE_INVALID_AMOUNT);
         }
+
+        verifyAmountIsWithinRange(amount);
+
         return amount;
     }
 
+    /**
+     * Throws a {@code ParseException} if the amount is out of range.
+     */
+    public static void verifyAmountIsWithinRange(double amount) throws ParseException {
+        double minAmount = -Math.pow(10, 9);
+        double maxAmount = Math.pow(10, 9);
```

Comment:
Technically fine but better practice might've been to make them a constant 1000000000